### PR TITLE
Johannes/end screen

### PR DESCRIPTION
**Pull request by holke**
Added stats that track the number of infected/exposed etc.

Print those stats on the end screen.

### CODE REVIEW

**Comment by holke** [comment]
Merged master updates into this branch.

File: YouVsVirus/Assets/Scripts/Components/LevelStats.cs [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+    /// Initialize with initial number of NPCs and initial number
+    /// of infected.
+    /// </summary>
+    public void Init(int InitialNumberOfNPCs, int NumberOfInitiallyInfected)
+    {
+        // Set initiali numbers of NPCs and infected humans
+        NumberOfNPCs = InitialNumberOfNPCs;
+        NumberInfected = NumberOfInitiallyInfected;
+        // At the beginning only the infected humans are exposed
+        NumberExposed = NumberOfInitiallyInfected;
```
Comment (ghost) [inline YouVsVirus/Assets/Scripts/Components/LevelStats.cs]: The initially infected are counted twice: Once in this line, and once in HumanBase's `Start()` method when `SetCondition(_initialCondition)` is called. I guess we solve this by deleting this line?

**Review by ghost — CHANGES_REQUESTED** [review]
Good work! There is one problem: When restarting the game ("Neustart" button), a new LevelStats  object is created. In the second game run, both objects exist twice - and it is not clear which one is returned by `GameObject.Find()`. 

Suggestion: Use the singleton mechanism I already implemented in LevelSettings for LevelStats, too, and remove the LevelStats and LevelSettings Game Objects from the MainMenu scene in the Unity Editor. This way, both GameObjects will be created programmatically upon first access, and their uniqueness will be guaranteed. We will have to make sure to completely reset the LevelStats upon restart. Also, replace every call to `GameObject.Find("LevelStats")` by the static singleton getter.

**Comment by holke** [comment]
Thanks for the suggestion with the singleton. Will change it soon.

File: YouVsVirus/Assets/Scripts/Components/LevelStats.cs (hunk 1 above)
Comment (holke) [reply YouVsVirus/Assets/Scripts/Components/LevelStats.cs]: Good catch!
I think its safe to remove the Infected once from the Init routine then.

Comment (holke) [reply YouVsVirus/Assets/Scripts/Components/LevelStats.cs]: I fixed it now.

**Comment by holke** [comment]
Implemented the singleton stuff now.
Seems to work fine.

**Review by maccxs — COMMENTED** [review]
Hi,
from my point of view it looks good and seems to be working. I did not have a look at the intrinsics since @FredyHennig already did. From a design point of view, I would like the panel to do not go down that far. I'd rather have the "Neustart" and "Beenden" button on the same height as the YouVsVirus text on the left (if need be that could go down a bit further). One more small thing I noticed which is also in the master: Somehow we introduced a small white line on the left side of the screen which should no be there both in the main and the end menu.

**Comment by holke** [comment]
Updated the look of the end screen.

@maccxs: I cannot see any white line here...or i do not recognize it.
Can you post a screenshot of what you mean?

**Comment by maccxs** [comment]
grafik

**Comment by holke** [comment]
Have you built and installed the game there or is this from unity?

Let´s divert this discussion to https://github.com/youvsvirus/youvsvirus-unity/issues/36

**Comment by maccxs** [comment]
This is from unity

**Comment by ghost** [comment]
The LevelStats Object was still preinitialized in the MainMenu Scene, so at every restart a new one would be created by Unity. Fixed that ;)

**Comment by ghost** [comment]
oops, that was the wrong button.

**Comment by holke** [comment]
Is this request done then and can be merged?

**Comment by ghost** [comment]
Yes, I think now we've got it all :)

**Review by ghost — APPROVED** [review]

**Pull request merged**